source/Octopus.Tentacle.Contracts/ScriptTicket.cs

```diff
@@ -33,7 +32,7 @@ public override bool Equals(object? obj)
         }
 
         public override int GetHashCode()
-            => TaskId != null ? TaskId.GetHashCode() : 0;
+            => TaskId != null ? StringComparer.OrdinalIgnoreCase.GetHashCode(TaskId) : 0;
```

Comment:
Is this backward compatible e.g. before and after the change return the same HashCode?

Reply:
No, it won't. 

But that was the bug I thought was worth considering fixing. 

Because to me, the script ticket `ServerTasks-12881-5e494d62-83fa-4229-9292-05f0fcfb91f5` should be the same as `SERVERTASKS-12881-5E494D62-83FA-4229-9292-05F0FCFB91F5`

This Linqpad query shows the differences

```
"A".GetHashCode().Dump();
"a".GetHashCode().Dump();
StringComparer.OrdinalIgnoreCase.GetHashCode("A").Dump();
StringComparer.OrdinalIgnoreCase.GetHashCode("a").Dump();
```
Returned this (i.e. case matters normally):
```
1715730010
574887235
1715730010
1715730010
```

But I'm ok to revert this if backwards compatibility is more important?

Reply:
I'm considering the edge case here of 

- Script Execution starts
- Tentacle is upgraded
- RPC Retries are enabled so Server "recovers from the Tentacle Restart"
- Tentacle reports the script is unknown instead of it finished/it was running but never completed (or tentacle didn't know it completed)

It's a small issue, an edge case and probably unlikely anyone will hit the issue so maybe ok?

Reply:
Just to clarify, are you saying we will hit this edge case if we fix the bug, or if we leave it in?

I think it might eventuate if we leave it in. This is because I don't think it affects the original `ScriptService`, as that uses a `ConcurrentDictionary` with `StringComparer.OrdinalIgnoreCase`. So it's just a bug for the new `ScriptServiceV2` (as that uses the domain object itself as the key).

But if we fix it, we fix `ScriptServiceV2` as well?

This might be moot, because I would have thought that whatever casing Octopus Server gives the script ticket would be consistent.

Also, Octopus Server would pass around this object as well. That makes me a little nervous about the change. But I still can't look past the obvious original bug... 😑